src/openrct2/network/NetworkUser.cpp

```diff
@@ -178,29 +164,19 @@ void NetworkUserManager::UnsetUsersOfGroup(uint8_t groupId)
 
 void NetworkUserManager::RemoveUser(const std::string& hash)
 {
-    NetworkUser* networkUser = GetUserByHash(hash);
+    NetworkUser* networkUser = const_cast<NetworkUser*>(GetUserByHash(hash));
```

Comment:
`const_cast` shouldn't be necessary, and sometimes dangerous to use for removing const. In this case, where the underlying container cannot be optimized away, it should be good, but why is `GetUserByHash` const at all?

If it's necessary to keep the const-variant around, the C++ way would be to add a non-const overload that then gets used here.

Reply:
`GetUserByHash` is const because it does not mutate the state of `NetworkUserManager`.
There were two versions of `GetUserByHash` at first (both const and non-const version) but I think it's just unnecessary duplication so removed non-const version.

Actually, according to the [guide](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#example-210), the most preferred way is to use template though ;)

Reply:
Since you need a non-cost overload IMO that means it is **not** unnecessary duplication.

AFAIK the common thing to do to reduce duplication is to use `const_cast` indeed, but not here - instead, cast constness away in a non-const overload calling the const overload. In this instance, you know for sure that the object is not const so casting constness away is harmless.

Reply:
Fair point overloading non const version with `const_cast`. I will prepare another PR for the fix.